src/devana/syntax_abstraction/functioninfo.py

```diff
@@ -10,84 +10,234 @@
+                        result.noexcept_value = other.noexcept_value
+                    else:
+                        raise CodeError("Idk")
+                elif self.noexcept_value is not None:
+                    result.noexcept_value = self.noexcept_value
+                else:
+                    result.noexcept_value = None
+            return result
+        elif isinstance(other, FunctionModification.ModificationKind):
+            result = FunctionModification(self.value | other)
+            if self.is_noexcept:
+                result.noexcept_value = self.noexcept_value
+            return result
+        raise NotImplementedError()
+
+    def __xor__(self, other):
+        if isinstance(other, type(self)):
+            result = FunctionModification(self.value.__or__(self.value, other.value))  # noqa
+            if result.is_noexcept:
+                if self.is_noexcept:
+                    result.noexcept_value = self.noexcept_value
+                else:
+                    result.noexcept_value = other.noexcept_value
+            return result
+        elif isinstance(other, FunctionModification.ModificationKind):
+            result = FunctionModification(self.value.__or__(self.value, other)) # noqa
+            if result.is_noexcept:
+                if self.is_noexcept:
+                    result.noexcept_value = self.noexcept_value
+            return result
+        raise NotImplementedError()
+
+    def __eq__(self, other):
+        if isinstance(other, FunctionModification):
+            if self.noexcept_value is not None and other.noexcept_value is not None:
+                return self.value == other.value and self.noexcept_value == other.noexcept_value
+            elif self.noexcept_value is None and other.noexcept_value is None:
+                return True
+            else:
+                return False
+        elif isinstance(other, FunctionModification.ModificationKind):
+            if self.noexcept_value is None:
+                return True
+        return False
+
+    def __invert__(self):
+        result = FunctionModification(~self.value)
+        return result
+
+    def __bool__(self):
+        return bool(self.value)
+
+    __ror__ = __or__
+    __rand__ = __and__
+    __rxor__ = __xor__
+
+    def __str__(self):
+        result = f"{str(self.value)}"
+        if self.value & FunctionModification.ModificationKind.NOEXCEPT:
+            result += f" (Noexcept value: {self.noexcept_value})"
+        return result
+
+    @property
+    def noexcept_value(self) -> Optional[str]:
+        return self._noexcept_value
+
+    @noexcept_value.setter
+    def noexcept_value(self, value):
+        if not self.value & FunctionModification.ModificationKind.NOEXCEPT:
+            if value is not None:
+                self.value |= FunctionModification.ModificationKind.NOEXCEPT # noqa
+        if value is None:
+            self.value &= ~FunctionModification.ModificationKind.NOEXCEPT  # noqa
```

Comment:
None is normal value for noexcept_value, when is noexcept but without txt so please don't remove NOEXCEPT from valuie in this case.